docs/reference/contributing/target/Watchdog.md

```diff
@@ -1,25 +1,45 @@
-<h3 id="watchdog-port">Watchdog</h3>
+### Watchdog
 
-[Include a brief description here.]
+The Watchdog HAL API provides a low-level interface to the Independent Watchdog Timer of a target. Implementing the Watchdog API is not mandatory, but by implementing it, you can use the hardware watchdog timer to detect and recover from computer malfunctions.
+
+Implement the Reset Reason API when implementing the Watchdog API. The Reset Reason API allows you to detect at boot time whether a watchdog caused the last reset.
```

Comment:
Query: Should there be a space in "Reset Reason API", or should it match the other APIs and be "ResetReason API"?

Reply:
Yeah, it should be ResetReason API.